pkg/client/delegation/client_integration_test.go

```diff
@@ -0,0 +1,214 @@
+package delegation_test
+
+// TODO(@h5law): Figure out how to use real components of the localnet
+//	- Create app and gateway actors
+//  - Stake them
+//  - Delegate to the gateway
+//  - Undelegate from the gateway
+// Currently this test doesn't work, because (I think) it is using a mock
+// keeper etc and this isnt actually interacting with the localnet where
+// the DelegationClient is listening for events from.
+
+import (
+	"testing"
+)
+
+func TestDelegationClient_DelegateeChangesObservable(t *testing.T) {
+	t.Skip("TODO")
+}
+
+// const (
+// 	delegationIntegrationSubTimeout = 15 * time.Second
+// )
+
+// func createNetworkWithApplicationsAndGateways(t *testing.T, numApplications, numGateways int) *network.Network {
+// 	// Prepare the network
+// 	cfg := network.DefaultConfig()
+// 	net := network.New(t, cfg)
+// 	ctx := net.Validators[0].ClientCtx
+//
+// 	// Prepare the keyring for the 2 applications and 1 gateway account
+// 	kr := ctx.Keyring
+// 	accounts := testutil.CreateKeyringAccounts(t, kr, 3)
+// 	ctx = ctx.WithKeyring(kr)
+//
+// 	// Initialise all the accounts
+// 	for i, account := range accounts {
+// 		signatureSequenceNumber := i + 1
+// 		network.InitAccountWithSequence(t, net, account.Address, signatureSequenceNumber)
+// 	}
+// 	// need to wait for the account to be initialized in the next block
+// 	require.NoError(t, net.WaitForNextBlock())
+//
+// 	addresses := make([]string, len(accounts))
+// 	for i, account := range accounts {
+// 		addresses[i] = account.Address.String()
+// 	}
+//
+// 	// Create two applications
+// 	appGenesisState := network.ApplicationModuleGenesisStateWithAccounts(t, addresses[0:2])
+// 	buf, err := cfg.Codec.MarshalJSON(appGenesisState)
+// 	require.NoError(t, err)
+// 	cfg.GenesisState[apptypes.ModuleName] = buf
+//
+// 	// Create a single gateway
+// 	gatewayGenesisState := network.GatewayModuleGenesisStateWithAccounts(t, addresses[2:3])
+// 	buf, err = cfg.Codec.MarshalJSON(gatewayGenesisState)
+// 	require.NoError(t, err)
+// 	cfg.GenesisState[gatewaytypes.ModuleName] = buf
+//
+// 	return net
+// }
+
+// TestDelegationClient_DelegateeChangesObservable tests that the DelegationClient
+// can subscribe to the DelegateeChange events and that the events contain
+// the correct AppAddress, it does so by simulating the delegation
+// and undelegation of two applications to a gateway.
+// func TestDelegationClient_DelegateeChangesObservables(t *testing.T) {
```

Comment:
You should be able to uncomment most of this and add a `t.SkipNow()` call to prevent it from running:
```suggestion
// TODO_TEST/TODO_TECHDEBT: ...
func TestDelegationClient_DelegateeChangesObservables(t *testing.T) {
	t.SkipNow()
```